### PR TITLE
Rspecdocs

### PR DESCRIPTION
**Pull request by andrewsouthard1**
Created tests for document model, commented out PAPERCLIP_STORAGE_OPTIONS in document model

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -26,7 +26,8 @@ gem 'coffee-rails', '~> 4.2'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
 gem 'paperclip'
-gem 'aws-sdk'
+# gem 'aws-sdk', '< 2.0'
```
Comment (trevordjones) [inline Gemfile]: double check on this - make sure we're using the officially supported gem

File: app/models/document.rb [hunk 2]
```diff
@@ -1,7 +1,12 @@
 class Document < ApplicationRecord
   belongs_to :application
 
-  has_attached_file :attachment, PAPERCLIP_STORAGE_OPTIONS 
+  has_attached_file :attachment
+  # , PAPERCLIP_STORAGE_OPTIONS
```
Comment (trevordjones) [inline app/models/document.rb]: Keep what was originally there - add a `PAPERCLIP_STORAGE_OPTIONS` constant in the `config/environments/test.rb` file

File: app/models/document.rb [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
 
-  has_attached_file :attachment, PAPERCLIP_STORAGE_OPTIONS 
+  has_attached_file :attachment
+  # , PAPERCLIP_STORAGE_OPTIONS 
   validates_attachment :attachment, content_type: { content_type: ['application/pdf', 'application/msword', 'application/xml', /\Aapplication\/vnd.openxmlformats-.*/, 'image/jpg', 'image/jpeg', 'image/png', 'image/gif'] }
   validates :attachment_file_name, presence: true
+
+  validates :category, inclusion: { in: ['health_form', 'personal_essay', 'recommendation', 'resume', 'toefl_ielts_score_report', 'transcript', 'wes_evaluation', 'contract', 'student_visa', 'passport_photo'] }
+
+  validates :attachment_file_size, numericality: { greater_than: 0 }
```
Comment (trevordjones) [inline app/models/document.rb]: Let's not add this validation

File: app/models/workshop.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-class Workshop < ApplicationRecord
+  class Workshop < ApplicationRecord
```
Comment (trevordjones) [inline app/models/workshop.rb]: Indent this back

File: spec/factories/documents.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  factory :document do 
+    association :application
+    category "health_form"
+    attachment_file_name "1099CompositeandYearEndSummary20160210176928.pdf"
+    attachment_file_size 73764
+    attachment_content_type "application/pdf"
+  end
+end
```
Comment (trevordjones) [inline spec/factories/documents.rb]: Make sure each factory is in its own file

File: spec/models/document_spec.rb [hunk 6]
```diff
@@ -1,5 +1,49 @@
 require 'rails_helper'
 
 RSpec.describe Document, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it "should have a valid factory girl" do
```
Comment (trevordjones) [inline spec/models/document_spec.rb]: Remove 'should', use something like `it "has a valid factory girl"`
http://www.betterspecs.org/#expect

File: spec/models/document_spec.rb [hunk 7]
```diff
@@ -1,5 +1,49 @@
...
+    document = FactoryGirl.create(:document)
+    document.attachment_file_size = 100
+    expect(document).to be_valid
+  end
+
+  it "can't have a file size zero or below" do
+    document = FactoryGirl.create(:document)
+    document.attachment_file_size = 0
+    expect(document).to_not be_valid
+  end
```
Comment (trevordjones) [inline spec/models/document_spec.rb]: Won't need this one anymore

File: Gemfile (hunk 1 above)
Comment (jameschou93) [inline Gemfile]: officially supported gem seems to be aws-sdk, '<~ 2'

https://github.com/aws/aws-sdk-ruby

File: spec/factories/documents.rb (hunk 5 above)
Comment (jameschou93) [inline spec/factories/documents.rb]: done

**Pull request closed without merging**